packages/core/src/renderer/components/kube-object-list-layout/kube-object-list-layout.tsx

```diff
@@ -233,6 +234,15 @@ class NonInjectedKubeObjectListLayout<
         {...layoutProps}
       />
     );
+
+    return (
+      <TableDataContext.Provider
+        value={{
+          columns: targetColumns as GeneralKubeObjectListLayoutColumn[],
+        }}>
+        {itemsListLayout}
+      </TableDataContext.Provider>
+    );
```

Comment:
Probably we don't need to use `React.createContext()` at all and better to pass `props.columns` into 2 levels of components to `lens/packages/core/src/renderer/components/item-object-list/content.tsx` (see example in demo branch)

Reply:
Lets make it as a next step though.

Reply:
Removed context and pass columns with a props.